extensions/funding/service.go

```diff
@@ -186,20 +73,31 @@ func deriveDIDs(data *clientfunpb.FundingData) ([]identity.DID, error) {
 	return c, nil
 }
 
-func (s service) DeriveFromPayload(ctx context.Context, req *clientfunpb.FundingCreatePayload, identifier []byte) (documents.Model, error) {
+func (s service) DeriveFromPayload(ctx context.Context, req *clientfunpb.FundingCreatePayload) (model documents.Model, err error) {
 	var fd Data
 	fd.initFundingFromData(req.Data)
 
-	model, err := s.GetCurrentVersion(ctx, identifier)
+	var docID []byte
+	if req.DocumentId != "" {
```

Comment:
maybe to avoid nested if statements:
```golang
if req.DocumentId == ""  {
return nil, documents.ErrDocumentIdentifier
}
docID, err = hexutil.Decode(req.DocumentId)
if err != nil {
   return nil, err
}
````

Reply:
We should rename the error to `ErrDocumentID`